test/integration/components/config.test.js

```diff
@@ -2,69 +2,55 @@
     assert.equal(pubnub.getAuthKey(), 'authKey2');
+    });
   });
 
-  describe('UUID storage', () => {
-    let database = () => {
-      let db = {};
-      return {
-        get: (key) => db[key],
-        set: (key, value) => { db[key] = value; }
+  describe('uuid Parameter', () => {
+    it('throws when not provided value', () => {
+      let config = {
+        subscribeKey: 'mySubKey',
+        publishKey: 'myPublishKey',
+        authKey: 'authKey1'
       };
-    };
-
-    it('uses the UUID if it is provided in setup', () => {
-      let networking = new Networking({ keepAlive, get, post, proxy });
-      let storageParams = { uuid: 'customUUID', networking: networking };
-      const pubnub = new PubNub(storageParams);
-      assert.equal(pubnub.getUUID(), 'customUUID');
+      assert.throws(() => {
+        new PubNub(config);
+      });
     });
 
-    it('generates the UUID if it is not provided', () => {
-      let networking = new Networking({ keepAlive, get, post, proxy });
-      let storageParams = { networking: networking };
-      const pubnub = new PubNub(storageParams);
-      assert.equal(lilUUID.isUUID(pubnub.getUUID().replace('pn-', '')), true);
+    it('get/set', () => {
+      let pubnub = new PubNub({
+        subscribeKey: 'mySubKey',
+        publishKey: 'myPublishKey',
+        uuid: 'uuid1'
+      });
+      assert.equal(pubnub.getUUID(), 'uuid1');
+      pubnub.setUUID('uuid2');
+      assert.equal(pubnub.getUUID(), 'uuid2');
     });
 
-    it('checks UUID from database if db object is provided', () => {
-      let dbInstance = database();
-      let networking = new Networking({ keepAlive, get, post, proxy });
-      sinon.spy(dbInstance, 'get');
-      sinon.spy(dbInstance, 'set');
-      let storageParams = { subscribeKey: 'mySubKey', db: dbInstance, networking: networking };
-      const pubnub = new PubNub(storageParams);
-      assert.equal(dbInstance.get.callCount, 1);
-      assert.equal(dbInstance.get.getCall(0).args[0], 'mySubKeyuuid');
-      assert.equal(lilUUID.isUUID(pubnub.getUUID().replace('pn-', '')), true);
-    });
-
-    it('uses UUID from database if db object is provided', () => {
-      let dbInstance = database();
-      let networking = new Networking({ keepAlive, get, post, proxy });
-      dbInstance.set('mySubKeyuuid', 'dbUUID');
-      sinon.spy(dbInstance, 'get');
-      sinon.spy(dbInstance, 'set');
-      let storageParams = { subscribeKey: 'mySubKey', db: dbInstance, networking: networking };
-      const pubnub = new PubNub(storageParams);
-      assert.equal(dbInstance.get.callCount, 2);
-      assert.equal(dbInstance.get.getCall(0).args[0], 'mySubKeyuuid');
-      assert.equal(pubnub.getUUID(), 'dbUUID');
+    it('throws when invalid value provided', () => {
+      let config = {
+        subscribeKey: 'mySubKey',
+        publishKey: 'myPublishKey',
+        uuid: ' '
+      };
+      assert.throws(() => {
+        new PubNub(config);
+      });
```

Comment:
@mohitpubnub should we have test case for invalid UUID (empty) set with `pubnub.setUUID`?

Reply:
Yes, Good to add more tests.